modules/web/src/app/node-data/component.ts

```diff
@@ -106,6 +108,8 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
   isLoadingOSProfiles: boolean;
   isEnterpriseEdition = DynamicModule.isEnterpriseEdition;
   wizardMode: WizardMode;
+  currentNodeOS: OperatingSystem;
+  allowedOperatingSystems: AllowedOperatingSystems = {};
```

Comment:
Just one problem here is that if settings API request fails or if `settings.allowedOperatingSystems` is empty/undefined then there won't be any allowed operating systems. So IMO we should initialize this with all OS in `ngOnInit`.
For example:
```
{
  ubuntu: true,
  flatcar: true,
  ...
}